replica/table.cc

```diff
@@ -2966,4 +2966,9 @@ bool table::requires_cleanup(const sstables::sstable_set& set) const {
     }));
 }
 
+future<> table::cleanup_tablet(locator::tablet_id) {
+    co_await flush();
+    // FIXME: Remove sstables
```

Comment:
another good reason, in addition to the one pointed in the commit msg, is to release commitlog segments

Reply:
Yes, and to be safe, we need to address https://github.com/scylladb/scylladb/issues/4734

Reply:
We could also fence commitlog replay (allow only mutations that happened after the tablet was returned to the node). I don't know if it's better (and I wouldn't want to add commitlog replay state to tablet metadata).